source/extensions/filters/common/fault/fault_config.cc

```diff
@@ -18,12 +18,13 @@ HeaderPercentageProvider::percentage(const Http::RequestHeaderMap* request_heade
     return percentage_;
   }
   const auto header = request_headers->get(header_name_);
-  if (header == nullptr) {
+  if (header.empty()) {
     return percentage_;
   }
 
   uint32_t header_numerator;
-  if (!absl::SimpleAtoi(header->value().getStringView(), &header_numerator)) {
+  // Client controlled. Using the first value is fine.
```

Comment:
(belatedly) I'm not sure this comment is going to be illuminating to someone reading the code.
Maybe something along the lines of "This header is cleared unless set by trusted clients, and they should not be sending multi-lined values"?
for the "client controlled" ones here, above and below?